fava/application.py

```diff
@@ -325,6 +326,17 @@ def api_add_document():
             return api_error('{} already exists.'.format(filepath))
 
         file.save(filepath)
+
+        if request.form.get('bfilename', None):
+            # TODO use next_statement_key to calculate the metadata key to
+            # support multiple statement keys
```

Comment:
We can just find the corresponding entry. Also this Beancount-file related logic _really_ shouldn't be in `application`